File: Bit Manipulation/0190/0190.py
def func_190(data, bits=32):
    r = 0
    for i in range(bits):
        # (data % 2) 有效位，0或1
        # (2 ** (bits - 1 - i))当前位反向之后转换为10进制
        r += (data % 2) * (2 ** (bits - 1 - i))
        # 十进制转二进制
        data = data // 2
        # 转换结束提取退出
        if data == 0:
            break
    return r


data = 13  # ...001011
print(func_190(data, bits=4))  # -> 110100...

File: Bit Manipulation/0190/0190.cpp
#include <iostream>
#include <cmath>

class Solution
{
private:
    /* data */
public:
    int func_0190(int data, int bits = 32);
    Solution(/* args */);
    ~Solution();
};

Solution::Solution(/* args */)
{
}

Solution::~Solution()
{
}
int Solution::func_0190(int data, int bits)
{
    int result = 0;
    for (int i = 0; i < bits; i++)
    {
        result += (data % 2) * pow(2, bits - 1 - i);
        data = data / 2;
        if (0 == data)
            break;
    }
    return result;
}

int main(int argc, char **argv)
{
    Solution ss;
    int data = 11;
    int bits = 4;
    std::cout << ss.func_0190(data, bits=bits) << std::endl;
    return 0;
}
